pkg/api/graph/deployment.go

```diff
@@ -21,8 +21,9 @@ type DeploymentPipelineMap map[*DeploymentConfigNode][]ImagePipeline
 // ImagePipeline represents a build, its output, and any inputs. The input
 // to a build may be another ImagePipeline.
 type ImagePipeline struct {
-	Image ImageTagLocation
-	Build *BuildConfigNode
+	Image       ImageTagLocation
+	ImageStream *ImageStreamNode
```

Comment:
That obviates the need for tracking the image stream separately - it's just a check on whether an outbound edge exists when you create the pipeline.